### ISSUE TITLE
publish  with ClickOnce

### ISSUE THREAD

**Issue by ctaggart** [robmikh/minesweeper-rs#27]
Hi @robmikh, @kennykerr suggested that I use minesweeper-rs as an example https://github.com/microsoft/winrt-rs/issues/384 after I asked about ClickOnce deployment of a WinRT/Rust example https://github.com/microsoft/winrt-rs/issues/372. I am interested in trying out ClickOnce, as well as publishing to winget and the Microsoft Store. For all of the options, it looks like I'll need a valid code signing certificate, so I ordered one for my Taggart Software LLC for $84. Unfortunately, it looks like it may be up to a week before I can get it validated by Sectigo due to the wrong phone number being listed at Dun & Bradstreet.

My goal is just to try out the various options for publishing a WinRT/Rust app and share that knowledge. I'll probably create a fork in either https://github.com/ctaggart or https://github.com/taggartsoftware in order to figure out how to do it. I'm creating this issue in case this is something you wish to collaborate. A png program icon would be super helpful. 😃

**Comment by robmikh** [comment]
That sounds cool! I'd be interested to see what the workflow looks like, being able to generate simple installers would be nice.

I have a [placeholder icon in the uwp branch](https://github.com/robmikh/minesweeper-rs/tree/uwp/assets), but it's nothing special. 

Let me know if there's anything you'd need from me or if you'd like any help! I'm fairly unfamiliar with ClickOnce.

**Comment by ctaggart** [comment]
I'm not super familiar with ClickOnce. I've used it a few times, but have not yet deployed an app with it. Years ago, I created a Java Web Start app. I think they are similar tech, but the docs about using ClickOnce with Visual C++ made me think it would be possible to use ClickOnce with a native app. I received my code signing cert finally, so I tried last night without luck. I opened up https://github.com/MicrosoftDocs/cpp-docs/issues/2671  to see if they really meant C++/CLI applications. 

There is a 2008 [article](https://blog.michaelckennedy.net/2008/08/25/clickonce-deployment-for-unmanaged-code-c-vb6-etc/) that talks about  launching the native app with a C# shim:
``` csharp
static void Main(string[] args) {
    try {
        Process.Start("TheRealApp.exe");
    } catch(Exception x) {
        string msg = "Error launch application:\n\n" + x;
        string cap = "Error Launching Application";
        MessageBox.Show(msg, cap, MessageBoxButtons.OK, MessageBoxIcon.Error);
    }
}
```
but I'm hoping that isn't required.
